test/cpp/api/serialize.cpp

```diff
@@ -501,25 +508,41 @@ TEST(SerializeTest, Optim_Adagrad) {
   is_optimizer_state_equal<AdagradParamState>(optim1.state(), optim1_2.state());
 }
 
-TEST(SerializeTest, SerializationShouldPreserveIteration_SGD) {
-  std::vector<torch::Tensor> parameters = {
-      torch::randn({2, 2}), torch::randn({3, 3})};
-
-  torch::optim::SGD optimizer(parameters, 1.0);
+TEST(SerializeTest, Optim_SGD) {
+  test_serialize_optimizer<SGD, SGDOptions, SGDParamState>(SGDOptions(1e-1));
```

Comment:
@yf225 this test is failing for SGD here: https://github.com/pytorch/pytorch/blob/fd1a4f18ee7d7e44c7711c7ff3a383e1d93e1c6a/test/cpp/api/serialize.cpp#L150-L151 is the condition in the link necessarily true in all cases?

Reply:
I am not entirely sure - it might be worth checking whether the Python version of the same code holds true or not.